romancal/tweakreg/astrometric_utils.py

```diff
@@ -175,6 +180,101 @@ def compute_radius(wcs):
     return radius, fiducial
 
 
+class _CatalogCacheSingleton:
+    _instance: _CatalogCacheSingleton | None = None
+    _initalized: bool = False
+
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+
+        return cls._instance
```

Comment:
This turns the object into a singleton meaning on any given process/thread one gets the same object. The singletons maybe different actual objects on different threads but this is dealt with below.